### PR TITLE
Yak shaving

### PR DESCRIPTION
**Pull request by biomunky**
exploring splitting things out to make testing and adding new resources easier? Maybe?

### CODE REVIEW

**Comment by rsyi** [comment]
HA YES YAK SHAVING. I love it. There are a lot of threads to pull here...

Yeah the dash bit isn't used -- I originally wanted to implement a dashboard to visualize metrics when pressing a hotkey in the skim search (so you could see all the metrics that are calculated over that table, quickly), but never got around to doing it (also not sure if there's any desire for this sort of thing), so prob best to just let it die. :)

**Comment by biomunky** [comment]
can you face putting these changes through your testing routine? If it works out and you're happy to merge then I will take a stab at some more refactoring and tests. I have a small amount of spare time for some Rust stuff and don't have any immediate plans for my own side-projects.

**Comment by rsyi** [comment]
yeah absolutely. i'll QA this now.

File: cli/src/filesystem.rs [hunk 1]
```diff
@@ -7,26 +7,38 @@ use std::{
...
         get_manifest_dirname(),
         get_metrics_dirname(),
-    ];
+    ]
+    .iter()
+    .map(|directory| {
+        fs::create_dir_all(Path::new(directory))
+            .unwrap_or_else(|_| panic!("Directory {:} was not created succesfully.", directory))
+    })
+    .collect()
```
Comment (rsyi) [inline cli/src/filesystem.rs:24]: love it. so much cleaner

**Comment by rsyi** [comment]
Alright. got around to running a full round of QA tests. looks good -- merging! Thanks @biomunky! :)

**Pull request merged**